tsl/src/remote/async.c

```diff
@@ -234,7 +234,33 @@ async_request_send_prepare(TSConnection *conn, const char *sql, int n_params)
 		elog(ERROR, "cannot create prepared statement name");
 
 	req = async_request_create(conn, sql, stmt_name, n_params, NULL, FORMAT_TEXT);
-	req = async_request_send_internal(req, ERROR);
+
+	if (req->state != DEFERRED)
+		elog(ERROR, "can't send async request in state \"%d\"", req->state);
+
+	if (remote_connection_is_processing(req->conn))
```

Comment:
@erimatnor I am not sure if this case is handled correctly. As far as I understand, the statement is not prepared in this code path and the `AsyncRequest` is returned in `DEFERRED` state.

I could not find any code that deals with a returned `AsyncRequest` in this state (i.e., does `async_request_wait_prepared_statement` can deal with it?). Do I miss something?

Reply:
You are right. Was too quick to adopt code from another place.